inbm/dispatcher-agent/dispatcher/fota/guid.py

```diff
@@ -28,7 +27,7 @@ def extract_guids(fw_tool: str, types: list[str]) -> list[str]:
         raise FotaError("Firmware Update Aborted: failed to list GUIDs: {}".format(str(err)))
     guids = _parse_guids(out, types)
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Robustness$</h2>**
**Suggestion:** Handle potential exceptions from the PseudoShellRunner.run method more explicitly for improved error diagnostics.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        try:
            (out, err, code) = runner.run(cmd)
            if code != 0:
                raise FotaError("Firmware Update Aborted: failed to list GUIDs: {}".format(str(err)))
        except Exception as e:
            logger.error(f"Unexpected error running command: {cmd}, error: {e}")
            raise FotaError(f"Unexpected error occurred: {e}")
```